### PR TITLE
Fetch15 twitter with base64 images

### PR DESCRIPTION
**Pull request by iory**
fetchでtweetするときにbase64の画像を読み込んで、画像つきでtweetすることを可能にする変更です。
https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/375
こちらのPRがupstreamのものです。

@tkmtnt7000
https://github.com/knorth55/jsk_robot/pull/327 こちらのPRでキッチンデモをやってみたときにtweetが画像つきで表示されるのかを確認してみてください。

### CODE REVIEW

**Comment by tkmtnt7000** [comment]
ありがとうございます．早速試してみます．

File: rostwitter/python/rostwitter/twitter.py [hunk 1]
```diff
@@ -1,18 +1,16 @@
 # originally from https://raw.githubusercontent.com/bear/python-twitter/v1.1/twitter.py  # NOQA
 
+import math
 import json as simplejson
 import requests
+from itertools import zip_longest
```
Comment (tkmtnt7000) [inline rostwitter/python/rostwitter/twitter.py]: This line is only for python3.
Maybe `from itertools import izip_longest` is for python2

**Review by tkmtnt7000 — CHANGES_REQUESTED** [review]

File: rostwitter/python/rostwitter/twitter.py [hunk 2]
```diff
@@ -57,35 +55,57 @@ def _request_url(self, url, verb, data=None):
...
 
-    def _post_update_with_reply(self, texts, in_reply_to_status_id=None):
-        for text in texts:
+    def _post_update_with_reply(self, texts, media_list=None,
+                                in_reply_to_status_id=None):
+        split_media_list = []
+        media_list = media_list or []
+        for i in range(0, int(math.ceil(len(media_list) / 4.0))):
+            split_media_list.append(media_list[i * 4:(i + 1) * 4])
+        for text, media_list in zip_longest(texts, split_media_list):
```
Comment (tkmtnt7000) [inline rostwitter/python/rostwitter/twitter.py:67]: This line is also only for python3. 
I use izip_longest and successfully run kitchen-demo.

**Review by tkmtnt7000 — CHANGES_REQUESTED** [review]

**Comment by tkmtnt7000** [comment]
I tested this feature with kitchen-demo. 
See https://twitter.com/fetch_jsk/status/1561611173945978880, https://twitter.com/fetch_jsk/status/1561611164689510400, https://twitter.com/fetch_jsk/status/1561611144800120832, ... etc.

File: rostwitter/python/rostwitter/twitter.py (hunk 1 above)
Comment (iory) [reply rostwitter/python/rostwitter/twitter.py]: Thanks! Fixed it. https://github.com/knorth55/jsk_3rdparty/pull/31/commits/8186633a7e9defb7ca3777624424250d8dc29b42

File: rostwitter/python/rostwitter/twitter.py (hunk 2 above)
Comment (iory) [reply rostwitter/python/rostwitter/twitter.py:67]: Thanks! Fixed it. https://github.com/knorth55/jsk_3rdparty/pull/31/commits/8186633a7e9defb7ca3777624424250d8dc29b42

**Comment by iory** [comment]
@tkmtnt7000 Thanks!

https://github.com/tkmtnt7000/jsk_robot/pull/7
I made a change to see it as a THREAD when tweeting.
It would be great if you could check again.

File: rostwitter/python/rostwitter/twitter.py [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
 import json as simplejson
 import requests
-from requests_oauthlib import OAuth1
-# https://stackoverflow.com/questions/11914472/stringio-in-python3
 try:
-    from StringIO import StringIO ## for Python 2
-except ImportError:
-    from io import StringIO ## for Python 3
+    from itertools import zip_longest
+except (ImportError, ModuleNotFoundError):
```
Comment (tkmtnt7000) [inline rostwitter/python/rostwitter/twitter.py]: Sorry `ModuleNotFoundError` may be also only for python3.
```suggestion
except ImportError:
```

**Review by tkmtnt7000 — CHANGES_REQUESTED** [review]

File: rostwitter/python/rostwitter/twitter.py (hunk 3 above)
Comment (iory) [reply rostwitter/python/rostwitter/twitter.py]: Thanks!

**Comment by tkmtnt7000** [comment]
I think it's time to merge this PR, so could you please add commit if you have something to update and when you have time? @iory

**Comment by iory** [comment]
Thanks. I added commit of upstream https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/375

**Comment by tkmtnt7000** [comment]
Thank you for updating. It looks good.

**Review by tkmtnt7000 — APPROVED** [review]

**Comment by iory** [comment]
@tkmtnt7000 Thanks for your review! Merge this PR.

**Pull request merged**